server/src/main/java/org/opensearch/search/profile/aggregation/AggregationProfiler.java

```diff
@@ -47,29 +45,24 @@
  */
 public class AggregationProfiler extends AbstractProfiler<AggregationProfileBreakdown, Aggregator> {
 
-    private final Map<List<String>, AggregationProfileBreakdown> profileBreakdownLookup = new HashMap<>();
+    private final Map<Aggregator, AggregationProfileBreakdown> profileBreakdownLookup = new HashMap<>();
 
     public AggregationProfiler() {
         super(new InternalAggregationProfileTree());
     }
 
+    /**
+     * This method does not need to be thread safe for concurrent search use case as well.
+     * The `AggregationProfileBreakdown` for each Aggregation operator is created in sync path when `preCollection` is
+     * called on the Aggregation collector instances during construction.
```

Comment:
lets fix the javadoc to use the `{@link AggregationProfileBreakdown}` semantics to refer to a class and `{@link org.opensearch.search.aggregations.BucketCollector#preCollection()}` for method